### PR TITLE
Fix approximated predict contribution.

### PR DESCRIPTION
**Pull request by trivialfis**
Add a prediction type for approximated interaction.

### CODE REVIEW

File: include/xgboost/learner.h [hunk 1]
```diff
@@ -36,7 +36,8 @@ enum class PredictionType : std::uint8_t {  // NOLINT
   kContribution = 2,
   kApproxContribution = 3,
   kInteraction = 4,
-  kLeaf = 5
+  kApproxInteraction = 5,
```
Comment (hcho3) [inline include/xgboost/learner.h:39]: Is this a new feature? I don't see the necessary change anywhere in the C++ codebase.

File: src/c_api/c_api_utils.h [hunk 2]
```diff
@@ -71,6 +70,7 @@ inline void CalcPredictShape(bool strict_shape, PredictionType type, size_t rows
     }
     break;
   }
+  case PredictionType::kApproxInteraction:
```
Comment (RAMitchell) [inline src/c_api/c_api_utils.h:73]: Should something be happening here?

**Review by RAMitchell — COMMENTED** [review]
Did approximate feature interactions get disabled somewhere in a past refactor?

File: include/xgboost/c_api.h [hunk 3]
```diff
@@ -740,15 +740,17 @@ XGB_DLL int XGBoosterPredict(BoosterHandle handle,
...
+ * \param c_json_config String encoded predict configuration in JSON format, with
+ *                      following available fields in the JSON object:
  *
- *    "type": [0, 5]
+ *    "type": [0, 6]
  *      0: normal prediction
  *      1: output margin
  *      2: predict contribution
- *      3: predict approxmated contribution
+ *      3: predict approximated contribution
```
Comment (RAMitchell) [inline include/xgboost/c_api.h:750]: This is technically a breaking change right?

**Comment by trivialfis** [comment]
@hcho3 @RAMitchell It's a bug I introduced during the refactoring of the prediction funciton.  I had an oversight that the `approximate` parameter is useful for feature interaction and there was no test for it.  This PR is just adding it back.  Since both changes are in the same release so it's not a breaking change.

**Review by hcho3 — APPROVED** [review]

File: src/c_api/c_api_utils.h (hunk 2 above)
Comment (trivialfis) [reply src/c_api/c_api_utils.h:73]: No, the output shape is exactly the same between shap and approx.

**Pull request merged**